sqlmesh/core/model/definition.py

```diff
@@ -934,6 +936,18 @@ def _data_hash_values(self) -> t.List[str]:
 
         return data_hash_values
 
+    @property
+    def _additional_metadata(self) -> t.List[str]:
+        additional_metadata = super()._additional_metadata
+
+        for statement in (*self.pre_statements, *self.post_statements):
+            if isinstance(statement, d.MacroFunc) and not isinstance(statement, d.MacroDef):
+                target_macro = macro.get_registry().get(statement.name)
```

Comment:
944-946 look exactly like above. Can we have a function for this?
```
def _is_metadata_statement(statement: exp.Expression) -> bool:
    if isinstance(statement, d.MacroDef):
        return True
    if isinstance(statement, d.MacroFunc):
        ...
    return False

Reply:
Also `macro.get_registry()` might not always be populated correctly (eg. Airflow). Should we use `self.python_env` instead?

Reply:
you might need to extend `Executable` with the metadata flag however.